pkg/operator/gcwatchercontroller/gcwatcher_controller.go

```diff
@@ -102,16 +122,24 @@ func (c *GarbageCollectorWatcherController) syncWorker(ctx context.Context, sync
 	}
 	requiredAlertsSet := sets.NewString(c.alertNames...)
 
-	prometheusClient, err := newPrometheusClient(ctx, c.configMapClient)
-	if err != nil && errors.IsNotFound(err) {
-		// Prometheus client when failed to instantiate should not result in error being generated. This makes sure we don't
-		// generate excessive events.
-		return nil
-	} else if err != nil {
-		return err
+	if !c.promConnectivity.useCachedClient {
+		// Don't create prometheus client on every sync unless explicitly asked, re-use from struct, this will also
```

Comment:
we can switch this comment to: `use cached only for testing` or soemthing like that

Reply:
I added a comment during the struct declaration but I added one more here. Let me know what you think of this comment.

Reply:
looks good, although it might be better to put it before the if, since it is negated